test/r_tree/r_tree_test.dart

```diff
@@ -213,22 +214,26 @@ main() {
             tree.insert(item);
           }
         }
-        assertTreeHeightValidity(tree);
+        assertTreeValidity(tree);
+
+        expect(tree.currentRootNode, isA<NonLeafNode<dynamic>>());
 
         var items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(2500));
 
         data.forEach((RTreeDatum item) {
           tree.remove(item);
         });
-        assertTreeHeightValidity(tree);
+        assertTreeValidity(tree);
 
         items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(0));
 
+        expect(tree.currentRootNode, isA<LeafNode<dynamic>>());
```

Comment:
This asserts that `RTree._resetRoot` resets a non-leaf to being a leaf in the one edge case.